pyNN/nest/standardmodels/electrodes.py

```diff
@@ -70,25 +81,51 @@ def _phase_correction(self, start, freq, phase):
         nest.SetStatus(self._device, {'phase': phase_fix})
 
     def _round_timestamp(self, value, resolution):
-        # subtraction by 1e-12 to match NEURON working
-        return round ((float(value)-1e-12)/ resolution) * resolution
+        return int(value/resolution+0.5) * resolution
+
+    def _check_step_times(self, times, amplitudes, resolution):
+        # ensure that all time stamps are non-negative
+        if not all(times>=0.0):
```

Comment:
I think this is less efficient than `(times >= 0.0).all()`, provided `times` is a NumPy array

Reply:
Changed as suggested.